perception/autoware_tensorrt_bevdet/src/bevdet_node.cpp

```diff
@@ -0,0 +1,342 @@
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { camera_info_callback(3, msg); });
+
+  sub_b_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_b/camera_info", rclcpp::QoS{1},
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { camera_info_callback(4, msg); });
+
+  sub_br_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_br/camera_info", rclcpp::QoS{1},
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { camera_info_callback(5, msg); });
+}
+
+void image_transport(std::vector<cv::Mat> imgs, uchar * out_imgs, size_t width, size_t height)
+{
+  uchar * temp = new uchar[width * height * 3];
+  uchar * temp_gpu = nullptr;
+  CHECK_CUDA(cudaMalloc(&temp_gpu, width * height * 3));
+
+  for (size_t i = 0; i < imgs.size(); i++) {
+    cv::cvtColor(imgs[i], imgs[i], cv::COLOR_BGR2RGB);
+    CHECK_CUDA(cudaMemcpy(temp_gpu, imgs[i].data, width * height * 3, cudaMemcpyHostToDevice));
+    convert_RGBHWC_to_BGRCHW(temp_gpu, out_imgs + i * width * height * 3, 3, height, width);
+  }
+  delete[] temp;
+  CHECK_CUDA(cudaFree(temp_gpu));
+}
+
+void TRTBEVDetNode::callback(
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_fl_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_f_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_fr_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_bl_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_b_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_br_img)
+{
+  cv::Mat img_fl, img_f, img_fr, img_bl, img_b, img_br;
+  std::vector<cv::Mat> imgs;
+  img_fl = cv_bridge::toCvShare(msg_fl_img, "bgr8")->image;
+  img_f = cv_bridge::toCvShare(msg_f_img, "bgr8")->image;
+  img_fr = cv_bridge::toCvShare(msg_fr_img, "bgr8")->image;
+  img_bl = cv_bridge::toCvShare(msg_bl_img, "bgr8")->image;
+  img_b = cv_bridge::toCvShare(msg_b_img, "bgr8")->image;
+  img_br = cv_bridge::toCvShare(msg_br_img, "bgr8")->image;
+
+  imgs.emplace_back(img_fl);
+  imgs.emplace_back(img_f);
+  imgs.emplace_back(img_fr);
+  imgs.emplace_back(img_bl);
+  imgs.emplace_back(img_b);
+  imgs.emplace_back(img_br);
+
+  image_transport(imgs, imgs_dev_, img_w_, img_h_);
+
+  // uchar *imgs_dev
+  sampleData_.imgs_dev = imgs_dev_;
+
+  std::vector<Box> ego_boxes;
+  ego_boxes.clear();
+  float time = 0.f;
+
+  bevdet_->doInfer(sampleData_, ego_boxes, time);
+
+  autoware_perception_msgs::msg::DetectedObjects bevdet_objects;
+  bevdet_objects.header.frame_id = "base_link";
+  bevdet_objects.header.stamp = msg_f_img->header.stamp;
+
+  box3DToDetectedObjects(ego_boxes, bevdet_objects, class_names_, score_thre_);
+
+  pub_boxes_->publish(bevdet_objects);
+}
+
+void TRTBEVDetNode::camera_info_callback(int idx, const sensor_msgs::msg::CameraInfo::SharedPtr msg)
+{
+  if (caminfo_received_[idx])
+    return;  // already received;  not expected to modify because of we init the model only once
```

Comment:
The comment `// already received;  not expected to modify because of we init the model only once` in `camera_info_callback` suggests a potential race condition. While the model initialization happens only once, it's still unsafe to rely on the absence of modifications from other threads. Consider using appropriate synchronization mechanisms like mutexes to guarantee thread safety and protect the `caminfo_received_` vector from concurrent access.